### PR TITLE
Extend icon plugin with title attribute

### PR DESCRIPTION
**Pull request by zweckj**
## Proposed change

SSIA

## Type of change

- [x] Spelling, grammar or other readability improvements (`current` branch).
- [ ] Adjusted missing or incorrect information in the current documentation (`current` branch).
- [ ] Added documentation for a new integration I'm adding to Home Assistant (`next` branch).
  - [ ] I've opened up a PR to add logos and icons in [Brands repository](https://github.com/home-assistant/brands).
- [ ] Added documentation for a new feature I'm adding to Home Assistant (`next` branch).
- [ ] Removed stale or deprecated documentation.

## Additional information

- Link to parent pull request in the codebase: 
- Link to parent pull request in the Brands repository: 
- This PR fixes or closes issue: fixes #

## Checklist

- [x] This PR uses the correct branch, based on one of the following:
  - I made a change to the existing documentation and used the `current` branch.
  - I made a change that is related to an upcoming version of Home Assistant and used the `next` branch.
- [x] The documentation follows the Home Assistant documentation [standards].

[standards]: https://developers.home-assistant.io/docs/documenting/standards

## Summary by CodeRabbit

- **New Features**
	- Enhanced `icon` Liquid tag to support an optional title for improved usability in Jekyll templates.
	- Updated La Marzocco integration documentation to include new icon details and clarify API interaction.

- **Documentation**
	- Improved clarity on La Marzocco integration requirements, including account setup and supported functionalities.
	- Expanded troubleshooting section with specific issues and solutions related to connection and real-time updates.

### CODE REVIEW

**Comment by zweckj** [comment]
I'm not sure where you demo normally that these things work, I just added them to lamarzocco for verification, but that needs to be removed.

File: plugins/icon.rb [hunk 1]
```diff
@@ -5,26 +5,28 @@ class Icon < Liquid::Tag
       def initialize(tag_name, args, tokens)
         super
         if args.strip =~ SYNTAX
-          @icon = Regexp.last_match(1).downcase
+          @icon = Regexp.last_match[:icon].downcase
+          @title = Regexp.last_match[:title]
```
Comment (frenck) [inline plugins/icon.rb:9]: Should we generate a title from the icon automatically if it isn't set?

E.g., `mdi:bluetooth` resulting in `Bluetooth`?

Comment (zweckj) [reply plugins/icon.rb:9]: I like that

Comment (zweckj) [reply plugins/icon.rb:9]: works!

File: source/_integrations/lamarzocco.markdown [hunk 2]
```diff
@@ -67,6 +67,9 @@ Use Bluetooth:
 
 # Available platforms & entities
 
+{% icon "mdi:home" title="Home Icon" %}
+{% icon "mdi:home" %}
```
Comment (frenck) [inline source/_integrations/lamarzocco.markdown]: Nice! But let's remove this one to prevent we merge it in 😉 

../Frenck

**Review by frenck — CHANGES_REQUESTED** [review]

File: source/_integrations/lamarzocco.markdown (hunk 2 above)
Comment (zweckj) [reply source/_integrations/lamarzocco.markdown]: done, just wanted you to see that it actually works 😉

**Review by frenck — APPROVED** [review]
Nice! Thanks, @zweckj 👍 

../Frenck

**Pull request merged**